Added `Accordion2D` to the `UI` sub module

antrikshmisri opened the pull request:
This PR adds the `Accordion2D` UI to the `UI` sub module. Please merge #460 before reviewing/merging this. Below is a screenshot for reference:-
![Screenshot (11)](https://user-images.githubusercontent.com/54466356/128708360-8388b136-f178-4c78-a3e4-45e2ea182d01.png).

antrikshmisri committed:
feat: added TreeNode2D to UI

antrikshmisri committed:
feat: added TreeNode2D, Tree2D to UI

antrikshmisri committed:
feat: example for Tree2D

antrikshmisri committed:
fix: code cleanup

antrikshmisri committed:
removed redundant files

antrikshmisri committed:
fix: code cleanup, reduced duplication, methods

antrikshmisri committed:
fix: rebased w.r.t UI restructuring

antrikshmisri committed:
fix: code clean-up

antrikshmisri committed:
test files

antrikshmisri committed:
feat: icon for TreeNode2D

antrikshmisri committed:
test files

antrikshmisri committed:
minor fixes

antrikshmisri committed:
feat: TreeNode2D parent auto resizing

antrikshmisri committed:
fix: fixed logic for auto resizing

antrikshmisri committed:
fix: increased indentation for better visuals

antrikshmisri committed:
feat: tutorial for Tree2D

antrikshmisri committed:
removed redundant files

antrikshmisri committed:
fix: pep8 issues

antrikshmisri committed:
feat: selected nodes attribute

antrikshmisri committed:
tests: added tests for Tree2D UI

antrikshmisri committed:
events recording

antrikshmisri committed:
fix: pep8 issues

antrikshmisri committed:
feat: added a select child method

antrikshmisri committed:
tests: added tests for TreeNode2D

antrikshmisri committed:
fix: pep8 issues

antrikshmisri committed:
fix: removed unused imports

antrikshmisri committed:
test example

antrikshmisri committed:
UI fixes, addressed reviews

antrikshmisri committed:
fix: updated the Tree2D Ui widgety

antrikshmisri committed:
feat: resize w.r.t largest child

antrikshmisri committed:
fix: fixed title panel resizing

antrikshmisri committed:
docs: added actor modifier using tree in tutorials

antrikshmisri committed:
fix: addressed comments

antrikshmisri committed:
fix: fixed the tree2d tutorial

antrikshmisri committed:
feat: added multiselct feature

antrikshmisri committed:
test: updated tests for TreeNode2D

antrikshmisri committed:
Merge branch 'master' into tree-ui

antrikshmisri committed:
Merge branch 'master' into tree-ui

antrikshmisri committed:
feat: allow UI to be dragged by text/icon

antrikshmisri committed:
Merge branch 'tree-ui' into accordion-ui

antrikshmisri committed:
feat: added accordion to UI

antrikshmisri committed:
feat: icon setter/getter, click callback on text

antrikshmisri committed:
Merge branch 'tree-ui' into accordion-ui

antrikshmisri committed:
fix: pep8

antrikshmisri committed:
fix: pep8 fixes in example

antrikshmisri committed:
updated bullet points node

pep8speaks commented:
Hello @antrikshmisri! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:

* In the file [`docs/tutorials/02_ui/viz_accordion.py`](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/docs/tutorials/02_ui/viz_accordion.py):

> [Line 90:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/docs/tutorials/02_ui/viz_accordion.py#L90): [W391](https://duckduckgo.com/?q=pep8%20W391) blank line at end of file

* In the file [`docs/tutorials/02_ui/viz_tree_actor_modifier.py`](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/docs/tutorials/02_ui/viz_tree_actor_modifier.py):

> [Line 152:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/docs/tutorials/02_ui/viz_tree_actor_modifier.py#L152): [E305](https://duckduckgo.com/?q=pep8%20E305) expected 2 blank lines after class or function definition, found 1
> [Line 171:54](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/docs/tutorials/02_ui/viz_tree_actor_modifier.py#L171): [W292](https://duckduckgo.com/?q=pep8%20W292) no newline at end of file



* In the file [`fury/ui/elements.py`](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py):

> [Line 3396:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3396): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3480:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3480): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)
> [Line 3508:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3508): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (85 > 79 characters)
> [Line 3748:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3748): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3752:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3752): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3756:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3756): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3828:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3828): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3836:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3836): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 3837:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3837): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (90 > 79 characters)
> [Line 3838:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3838): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 3840:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3840): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3889:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3889): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3898:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3898): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 3900:80](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3900): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 3904:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3904): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3959:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3959): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3962:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3962): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3975:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3975): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3978:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3978): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 3990:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L3990): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4001:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4001): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4005:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4005): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4016:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4016): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4020:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4020): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4031:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4031): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace
> [Line 4035:1](https://github.com/fury-gl/fury/blob/42fe5556d553b87a134c71c7fa3966bb934eb3b2/fury/ui/elements.py#L4035): [W293](https://duckduckgo.com/?q=pep8%20W293) blank line contains whitespace

antrikshmisri committed:
fix: pep8 issues

antrikshmisri committed:
Merge branch 'tree-ui' into accordion-ui

antrikshmisri committed:
Merge branch 'master' into accordion-ui

antrikshmisri committed:
fix: fixed indentaion of child nodes

antrikshmisri committed:
feat: structure as a property

antrikshmisri committed:
test: added tests for Accordion UI

antrikshmisri committed:
fix: pep8 fixes

antrikshmisri committed:
feat: added second tutorial with text content

antrikshmisri committed:
fix: auto-resizing, label_text coords

antrikshmisri committed:
Merge branch 'tree-ui' into accordion-ui

antrikshmisri committed:
fix: removed redundant files

antrikshmisri committed:
Merge branch 'tree-ui' into accordion-ui

skoudoro commented on docs/tutorials/02_ui/viz_accordion.py:
```diff
@@ -0,0 +1,89 @@
+interactive = False
+
+if interactive:
+    show_manager.start()
```
Can you add a `record` at the end? we need a screenshot for this tutorial.

antrikshmisri replied:
Done

skoudoro commented on docs/tutorials/02_ui/viz_accordion_text_ui.py:
```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+"""
+================================================
+Example to demonstrate the use of Accordion2D UI
```
What is the difference from the previous one? You need to add more information to the description.

skoudoro commented on docs/tutorials/02_ui/viz_accordion_text_ui.py:
```diff
@@ -0,0 +1,67 @@
+interactive = False
+
+if interactive:
+    show_manager.start()
```
same comment as above. Add `window.record(....)`

antrikshmisri replied:
Done

skoudoro reviewed (commented):
Hi @antrikshmisri,

Can you rebase this PR and address my small comments. After that, I will have a deeper look and come back to you.

Thank you.

skoudoro commented:
Also, there is some pep8 issues, it would be great if you could address them. thank you

Commit:
Merge branch 'master' into accordion-ui
* master: (1622 commits)
  Test scipy import fix
  finalize
  remove scene from window in snapshot
  tests: leak on test_record
  fix: iren shouldn't be created inside of record method
  fix: double render call with timeline obj causes a seg fault
  updated glTF tutorials
  Let them loop
  Updated glTF animation tests
  Renamed some timelines leftovers
  Migrated glTF anim to the new animation sys
  Altered current_timestamp of single animations
  update sphinx team version
  lowercase
  updating videos with lower res
  removing extra comma
  skip marker_actor on linux
  skip test_timer
  fix some vtkerror
  update requirements
  ...

Commit:
Resolve comments on Accordion2D examples

+ Add more description to the Accordion2D examples
+ Add window.record to the Accordion2D examples

Commit:
Merge conflict resolution

+ Add imports that were removed in the previous merge conflict
  resolution.

antrikshmisri commented:
Hey @skoudoro, sorry for the late response (was tied up with some work). I have addressed all the latest comments. The `pep` errors are from the `Tree` UI for which there is a separate PR (https://github.com/fury-gl/fury/pull/460). Request you to review that PR first so that upon merge those changes will go away from this PR. Thanks!

skoudoro commented:
ok, thank you for the feedback @antrikshmisri, I will look at #460 first
